csharp/src/Apache.Arrow.Flight/Server/IFlightServer.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Grpc.Core;
+
+namespace Apache.Arrow.Flight.Server
+{
+    public interface IFlightServer
```

Comment:
What do you think about changing this to an `abstract class` instead, where all the base `virtual` methods just `throw new NotSupportedException()`?

The advantage is that:

1. If a service doesn't want to support something, they don't need to implement the method.
2. (more importantly) If/when new methods get added, we can add them to the base class without a binary breaking change.

Reply:
Good suggestion, changed to an abstract class and virtual methods.